### PR TITLE
Avoid browser autofill password recovery username

### PR DESCRIPTION
**Pull request by julien-nc**
This is an attempt to inhibit browser autofill for the username field in password recovery form. The field is now readonly until it gets user focus.

Apparently browsers ignore the "autocomplete" attribute.

### CODE REVIEW

**Comment by skjnldsv** [comment]
I'm really not fond of this hack to be honest.
Yes browsers ignores the autocomplete attribute, but then this is up to the browsrs to fix it. Let's not clutter our code for such papercut i'd say.

Usually it's causing issues because we're using generic type-text IDs/names, like `id="user" name="user"` iirc.
Changing to a less generic string might lead to better results.

**Comment by skjnldsv** [comment]
On a second though, why is this an issue btw?
I would find autocompletion quite relevant on a password reset form, no? (github is doing it for example, Google isn't though)

Here is what is working on google:

```html
<input type="email" autocomplete="off" spellcheck="false"
	aria-label="Phone or e-mail" name="identifier"
	value="" autocapitalize="none" id="recoveryIdentifierId">
```

**Comment by eneiluj** [comment]
> why is this an issue btw?

Browser can suggest unrelated user names I guess. Users might think the suggestion comes from Nextcloud and not from the browser itself so they might trust it provides a good value instead of trying to type their actual user name. I'm not strongly confident my explanation is realistic :grin:.

So the question becomes: Do we want to avoid browser autofill?

> I'm really not fond of those hacks to be honest.

@skjnldsv Same for me. I was hoping someone had a better idea. So if we decide to inhibit autofill, would you be ok if I change the id and name attributes to "recovuser"?

**Comment by skjnldsv** [comment]
> So the question becomes: Do we want to avoid browser autofill?

For me I think it's fine. :)

**Review by rosito1 — COMMENTED** [review]
- redacted -

File: core/src/components/login/ResetPassword.vue [hunk 1]
```diff
@@ -102,6 +104,7 @@ export default {
 			loading: false,
 			message: undefined,
 			user: this.username,
+			usernameReadonly: true,
```
Comment (rosito1) [inline core/src/components/login/ResetPassword.vue:107]: erdy_erdy_

File: core/src/components/login/ResetPassword.vue [hunk 2]
```diff
@@ -30,7 +30,9 @@
 					autocapitalize="off"
```
Comment (rosito1) [inline core/src/components/login/ResetPassword.vue:30]: https://instagram.com/erdy_erdy_?igshid=gop9gj69a51j

File: core/src/components/login/ResetPassword.vue [hunk 3]
```diff
@@ -30,7 +30,9 @@
...
 					:placeholder="t('core', 'Username or email')"
 					:aria-label="t('core', 'Username or email')"
+					:readonly="usernameReadonly"
 					required
+					@focus="usernameReadonly = false"
 					@change="updateUsername">
 				<!--<?php p($_['user_autofocus'] ? 'autofocus' : ''); ?>
 				autocomplete="<?php p($_['login_form_autocomplete']); ?>" autocapitalize="none" autocorrect="off"-->
```
Comment (rosito1) [inline core/src/components/login/ResetPassword.vue:38]: https://instagram.com/erdy_erdy_?igshid=op6k4vgout1g

File: core/src/components/login/ResetPassword.vue (hunk 1 above)
Comment (rosito1) [inline core/src/components/login/ResetPassword.vue:107]: erdy_erdy_

**Review by rosito1 — CHANGES_REQUESTED** [review]

**Review by rosito1 — COMMENTED** [review]
https://instagram.com/erdy_erdy_?igshid=op6k4vgout1g

**Comment by eneiluj** [comment]
@skjnldsv What do you think about emptying the username/email field when clicking on "forgot password"? This ensures the placeholder is shown and the user knows what the field is about.

**Comment by skjnldsv** [comment]
> @skjnldsv What do you think about emptying the username/email field when clicking on "forgot password"? This ensures the placeholder is shown and the user knows what the field is about.

Not fond of, not having to type your username again is a nice flow.

**Comment by jancborchardt** [comment]
@eneiluj since you closed this, should we delete the branch too? :)

**Pull request closed without merging**